CardScan/Classes/CreditCardOcr/MainLoopStateMachine.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  MainLoopStateMachine.swift
+//  CardScan
+//
+//  Created by Sam King on 8/5/20.
+//
+
+import Foundation
+
+public enum MainLoopState {
+    case initial
+    case ocrOnly
+    case cardOnly
+    case ocrAndCard
```

Comment:
These pertain to liveness only, right? Do we need them for cardscan?

Reply:
right, we'll use these for verify and liveness

Reply:
ahh, so you're planning to share this `MainLoopState` between the three? Would it make more sense to have an individual version of `MainLoopState` for each? It's confusing to have states that can never be reached.

Reply:
we could but you can't inherit from enums so the cleanest way would be to make it a class and inherit but I'm open to other suggestions since I do agree with you in principle (but I'm also ok with having it the way that it is currently).

Reply:
in Android, I specifically created separate implementations of a state machine for each product. I started with inheritance, but found that it didn't bring a lot of benefit and added a lot of complexity in implementation.

This is a relatively minor concern, so I don't consider it blocking, but I would personally create a separate state enum and main loop for each product.